src/dp_util.c

```diff
@@ -194,7 +194,7 @@ int dp_set_vf_rate_limit(uint16_t port_id, uint64_t rate)
 
 	if (res < 0 || (unsigned)res >= (int)sizeof(filename)) {
 		DPS_LOG_ERR("Failed to generate SR-IOV sysfs path or SR-IOV sysfs path to vf's max tx rate is too long");
-		return DP_ERROR;
+		return DP_GRPC_ERR_NOT_FOUND; // on Linux 6.x.y, this path/file does not seem to exist. thus for now, a not found error is returned and ignored by the caller
```

Comment:
This should be -ENOENT to emulate what DPDK does for missing entries. GRPC errors are using special values tied to metalnet/golang.

Also this is not the right place, this is `snprintf` failure, the right place is few lines down where `fopen` fails, in which case you can actually just return `-errno` like DPDK does.

Though seeing the code itself, I don't think you need to even differentiate because unless `fopen` fails, the only error is calling this function on invalid port (will fail down the line anyway) or `snprintf` failing (i.e. path is too long) which is actually again just a sysfs path problem.

Reply:
fixed.